_sources/iteration/mixed_up_code.rst

```diff
@@ -0,0 +1,319 @@
+    :numbered: left
+    :adaptive:
+    :noindent:
+    :practice: T
+
+    The program below should count down from 100 to 0 in decrements of
+    10 but the code is mixed up and contains extra blocks. Put the necessary blocks
+    in the correct order.
+    -----
+    int main() {
+    =====
+       int n = 100;
+    =====
+       int n = 10; #distractor
+    =====
+       while (n >= 0) {
+    =====
+       while (n < 0) { #distractor
+    =====
+       while (n > 0) { #distractor
+    =====
+          cout << n << endl;
+    =====
+          n -= 10;
+    =====
+          n += 10;                 #distractor
+    =====
+       }
+    =====
+    }
+
+.. parsonsprob:: iteration_mixed_up_code_3
+    :numbered: left
+    :adaptive:
+    :noindent:
+    :practice: T
+
+    The program below should find the sum of the first 10 natural numbers,
+    but the code is mixed up and contains extra blocks. Put the necessary blocks
+    in the correct order.
+    -----
+    int main() {
+    =====
+       int n = 10; #distractor
+    =====
+       int sum = 0;
+    =====
+       int sum = n;  #distractor
+    =====
+       for (int n = 1; n <= 10; ++n) {
+    =====
+       for (int n = 1; n < 10; ++n) {  #paired
+    =====
+       while (n < 100) {  #distractor
+    =====
+       while (n <= 9) {  #distractor
+    =====
+          cout << n << endl;  #distractor
+    =====
+          sum = sum + n;
+    =====
+       }
+    =====
+        cout << sum << endl;
+    =====
+    }
+
+.. parsonsprob:: iteration_mixed_up_code_4
+    :numbered: left
+    :adaptive:
+    :practice: T
+
+    Let's write the a program that uses a for
+    loop to print out "hello" three times.
+    -----
+    int main() {
+    =====
+       int n = 0;  #distractor
+    =====
+       for (int i = 0; i < 3; ++i) {
+    =====
+       for (int i = 1; i < 3; ++i) {                #paired
+    =====
+          cout << "hello" << endl;
+    =====
+          --i;  #distractor
+    =====
+       }
+    =====
+    }
+
+.. parsonsprob:: iteration_mixed_up_code_5
+    :numbered: left
+    :adaptive:
+
+    Now let's write the same program as the previous question but using a while loop.
+    -----
+    int main() {
+    =====
+       int n = 0;
+    =====
+       while (n < 3) {
+    =====
+       while (n <= n) {                        #paired
+    =====
+          cout << "hello" << endl;
+    =====
+          cout << hello << endl;                        #paired
+    =====
+          n++;
+    =====
+       }
+    =====
+    }
+
+.. parsonsprob:: iteration_mixed_up_code_6
+    :numbered: left
+    :adaptive:
+
+    Let's write a program that prints out all numbers from 1 to ``i``, with ``i`` going from 5 to 1.
+    For example, the first line should print "1 2 3 4 5", the second line should print "1 2 3 4" and so on until "1".
+    Use a nested for loop.
+    
+        ::
+        The ouput should be:
+        1 2 3 4 5
+        1 2 3 4
+        1 2 3
+        1 2 
+        1
```

Comment:
The formatting of lines 163-168 doesn't show up well. Also "output" is line 163 is misspelled.